CMakeLists.txt

```diff
@@ -26,15 +26,17 @@ set(USE_IBVERBS_DEFAULT OFF)
 set(USE_NCCL_DEFAULT OFF)
 set(USE_RCCL_DEFAULT OFF)
 set(USE_LIBUV_DEFAULT OFF)
-set(USE_TCP_OPENSSL_DEFAULT OFF)
+set(USE_TCP_OPENSSL_LINK_DEFAULT OFF)
+set(USE_TCP_OPENSSL_LOAD_DEFAULT OFF)
 
 # Options
 option(USE_REDIS "Support using Redis for rendezvous" ${USE_REDIS_DEFAULT})
 option(USE_IBVERBS "Support ibverbs transport" ${USE_IBVERBS_DEFAULT})
 option(USE_NCCL "Support using NCCL for local collectives" ${USE_NCCL_DEFAULT})
 option(USE_RCCL "Support using RCCL for local collectives" ${USE_RCCL_DEFAULT})
 option(USE_LIBUV "Build libuv transport" ${USE_LIBUV_DEFAULT})
-option(USE_TCP_OPENSSL "Build tcp+openssl transport" ${USE_TCP_OPENSSL_DEFAULT})
+option(USE_TCP_OPENSSL_LINK "Build tcp+openssl transport" ${USE_TCP_OPENSSL_LINK_DEFAULT})
+option(USE_TCP_OPENSSL_LOAD "Build tcp+openssl transport" ${USE_TCP_OPENSSL_LOAD_DEFAULT})
```

Comment:
Nits: Can you change message to "Enable SSL TLS support"  to one and "Enable SSL/TLS dynamic binding to another". 
Also, if they are mutually exclusive, can you check for that?

Reply:
addressed